### PR TITLE
NEXT-15802 - Build administration and storefront without database

### PR DESCRIPTION
**Pull request by shyim**

### CODE REVIEW

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Building assets of administration and storefront without a database
+
+ It is common to prebuilt assets in professional deployments to deploy the finished assets to the production environment. This task is mostly done by an CI job which doesn't have access to the production database. Shopware needs access to database to look up the installed extensions / loading the configured theme variables. To be able to build the assets without a database, we can use static dumped files.
+ This guide works requires Shopware 6.4.4.0 or higher.
+
+
+ # Compiling the Administration without database
+
+ By default, without a database Shopware considers only the Administration to be built. To include the extensions without a database, you will need to use the `ComposerPluginLoader`. This determines the used plugins by looking up into the project installed dependencies. To get this working the plugin needs to be required in the system using `composer req [package/name]`.
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:9]:
```suggestion
By default, without a database Shopware considers only the Administration to be built. To include the extensions without a database, you will need to use the `ComposerPluginLoader`. This determines the used plugins by looking up into the project installed dependencies. To get this working the plugin needs to be required in the system using `composer req [package/name]`.
```

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+ To use the `ComposerPluginLoader` you have to create a file like `bin/ci` and set up the CLI application with loader. There is an example:
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:11]:
```suggestion
 To use the `ComposerPluginLoader` you have to create a file like `bin/ci` and set up the CLI application with the loader. There is an example:
```

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+$pluginLoader = new ComposerPluginLoader($classLoader, null);
+$kernel = new HttpKernel($env, $debug, $classLoader);
+$kernel->setPluginLoader($pluginLoader);
+$application = new Application($kernel->getKernel());
+$application->run($input);
+```
+
+We can now dump the plugins for the administration with the new file without a database with the command `bin/ci bundle:dump`. It is recommended to call `bin/ci` instead of `bin/console` in the `bin/*.js` scripts.
+
+ # Compiling the Storefront without database
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:58]:
```suggestion
# Compiling the Storefront without database
```

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+ To compile the Storefront theme, we will need theme variables from the database. To allow compiling it without a database, it is possible to dump the variables to the private file system of Shopware. This file system interacts by default with the local folder `files`, but to compile it [should be shared with a storage adapter like s3](../../infrastructure/filesystem.md). The configuration can be dumped using the command `bin/console theme:dump`, or it happens automatically when changing theme settings, assigning a new theme.
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:60]:
```suggestion
To compile the Storefront theme, we will need the theme variables from the database. To allow compiling it without a database, it is possible to dump the variables to the private file system of Shopware. This file system interacts with the local folder `files` by default, but to compile it, [it should be shared with a storage adapter like s3](../../infrastructure/filesystem.md). The configuration can be dumped using the command `bin/console theme:dump`, or it happens automatically when changing theme settings or assigning a new theme.
```

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+ By default, Shopware still tried to load configuration from database, in next step we will need to change the loader to `StaticFileConfigLoader`. To change that we will need to create a new file `config/packages/storefront.yaml` with the following content:
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:62]:
```suggestion
By default, Shopware still tried to load configurations from the database, in the next step we will need to change the loader to `StaticFileConfigLoader`. To change that we will need to create a new file `config/packages/storefront.yaml` with the following content:
```

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+ ```yaml
+storefront:
+    theme:
+        config_loader_id: Shopware\Storefront\Theme\ConfigLoader\StaticFileConfigLoader
+        available_theme_provider: Shopware\Storefront\Theme\ConfigLoader\StaticFileAvailableThemeProvider
+ ```
+
+ This will force the theme compiler to use the static dumped file instead looking up in the database.
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:71]:
```suggestion
 This will force the theme compiler to use the static dumped file instead of looking into the database.
```

**Review by PaddyS — CHANGES_REQUESTED** [review]

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+$pluginLoader = new ComposerPluginLoader($classLoader, null);
+$kernel = new HttpKernel($env, $debug, $classLoader);
+$kernel->setPluginLoader($pluginLoader);
+$application = new Application($kernel->getKernel());
+$application->run($input);
+```
+
+We can now dump the plugins for the administration with the new file without a database with the command `bin/ci bundle:dump`. It is recommended to call `bin/ci` instead of `bin/console` in the `bin/*.js` scripts.
+
+# Compiling the Storefront without database
```
Comment (PaddyS) [inline guides/hosting/installation-updates/deployments/databaseless.md:58]:
```suggestion
## Compiling the Storefront without database
```

**Review by PaddyS — CHANGES_REQUESTED** [review]

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Building assets of administration and storefront without a database
+
+It is common to prebuilt assets in professional deployments to deploy the finished assets to the production environment. This task is mostly done by an CI job which doesn't have access to the the production database. Shopware needs access to the database to look up the installed extensions / loading the configured theme variables. To be able to build the assets without a database, we can use static dumped files.
+ This guide requires Shopware 6.4.0 or higher.
```
Comment (elkmod) [inline guides/hosting/installation-updates/deployments/databaseless.md:4]: Could you please mention the version requirement at the top of the article (right below the first heading) as described in this article: https://developer.shopware.com/docs/resources/guidelines/documentation/writing#maintaining-versions

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
+# Building assets of administration and storefront without a database
+
+It is common to prebuilt assets in professional deployments to deploy the finished assets to the production environment. This task is mostly done by an CI job which doesn't have access to the the production database. Shopware needs access to the database to look up the installed extensions / loading the configured theme variables. To be able to build the assets without a database, we can use static dumped files.
```
Comment (elkmod) [inline guides/hosting/installation-updates/deployments/databaseless.md:3]: by "a" CI job [...].

File: guides/hosting/installation-updates/deployments/databaseless.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+ This guide requires Shopware 6.4.0 or higher.
+
+
+## Compiling the Administration without database
+
+By default, without a database Shopware considers only the Administration to be built. To include the extensions without a database, you will need to use the `ComposerPluginLoader`. This determines the used plugins by looking up into the project installed dependencies. To get this working the plugin needs to be required in the system using `composer req [package/name]`.
```
Comment (elkmod) [inline guides/hosting/installation-updates/deployments/databaseless.md:9]: Maybe rephrase to:

By default, Shopware only builds the Administration without extensions if there is no database connection.

**Review by elkmod — APPROVED** [review]
